color-identifiers-mode.el

```diff
@@ -748,11 +748,15 @@ major mode, identifiers are saved to
 
 (defun color-identifiers:refontify ()
   "Refontify the buffer using font-lock."
-  (when font-lock-mode
-    (if (fboundp 'font-lock-flush)
+  (if (fboundp 'font-lock-flush)
+      (progn
         (font-lock-flush)
-      (with-no-warnings
-        (font-lock-fontify-buffer)))))
+        ;; Remove unnecessary indirection by making the function an alias. So this
+        ;; whole body effectively gets called just once.
+        (defalias 'color-identifiers:refontify 'font-lock-flush))
```

Comment:
Trying to alias the not-yet-defined containing function is what confuses the byte compiler. I'd suggest:
```el
(if (fboundp 'font-lock-flush)
    (defalias 'color-identifiers:refontify 'font-lock-flush)
  (defun color-identifiers:refontify ()
    "Refontify the buffer using font-lock.
    (with-no-warnings
      (and font-lock-mode (font-lock-fontify-buffer)))))
```

Reply:
Aaah, actually, I just remember a work around to that.

Your solution doesn't work, it still prints a `color-identifiers-mode.el:684:4: Error: the function ‘color-identifiers:refontify’ is not known to be defined.`. But the reason to that is [this Emacs bug](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=69120). And one work around to that is adding a `declare-function` right before the `defun`. [Like this](https://github.com/ankurdave/color-identifiers-mode/pull/122/commits/c233df377e95fb6c577856635dbf3ed87ea5709d).

Reply:
Huh, weird. I see code with that pattern all over the place, but have not seen that compilation error before.

Reply:
Well, maybe in other places the callers were in separate `.el` files? Because the error appears for the call-site, not the definition-site.

Reply:
No, not at all. Mostly the same .el file. But looking again at this, you're apparently getting an error at a call site on line 684, but defining the function on a later line ~750, so of course the function is not known to be defined at the earlier call site. Just move the definition earlier in the file.

Reply:
> But looking again at this, you're apparently getting an error at a call site on line 684, but defining the function on a later line ~750, so of course the function is not known to be defined at the earlier call site. Just move the definition earlier in the file.

I tried that too, it didn't help.

Reply:
…and as a matter of fact, adding `declare-function` works around the problem even though `declare-function` is called too on line ≈750. I presume function definitions are global to the file, so the order doesn't matter.

Reply:
This works too, and is a little neater IMO because it brings the `fboundp` check to compile time so that it is unambigous:
```el
(if (eval-when-compile (fboundp 'font-lock-flush))
    (defalias 'color-identifiers:refontify 'font-lock-flush)
  (defun color-identifiers:refontify ()
    "Refontify the buffer using font-lock."
    (with-no-warnings
      (and font-lock-mode (font-lock-fontify-buffer)))))
```

Reply:
It still really doesn't see *(I also tried moving the code to the top, it doesn't influence anything)*:

![image](https://github.com/user-attachments/assets/20417e20-58f9-45c0-9fed-09f6885dacb1)

Reply:
Right, yeah, was testing while compiling multiple times in the same instance. I found another pattern I've used in the past btw:
```
(defalias 'windswap--swap-states
  (if (fboundp 'window-swap-states)
      'window-swap-states
    (lambda (a b)
      "Docstring for local function definition.
      ...)))
```

Reply:
I probably found and solved that problem myself at some point in order to end up with that solution. :)